### PR TITLE
Fix test linting

### PR DESCRIPTION
**Pull request by arturo-seijas**

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -40,7 +41,7 @@ def test_misconfigured_charm_reaches_blocked_status():
     """
     harness = Harness(SamlIntegratorOperatorCharm)
     harness.begin()
-    harness.model.unit.status == ops.BlockedStatus()
+    assert harness.model.unit.status.name == ops.BlockedStatus().name
```
Comment (jdkandersson) [inline tests/unit/test_charm.py:44]: Do we need an instance or is the name available directly on the class?

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -51,11 +52,11 @@ def test_charm_reaches_active_status(urlopen_mock):
     assert: the charm reaches ActiveStatus.
     """
     with open("tests/unit/files/metadata_1.xml", "rb") as metadata:
-        cm = MagicMock()
-        cm.getcode.return_value = 200
-        cm.read.return_value = metadata.read()
-        cm.__enter__.return_value = cm
-        urlopen_mock.return_value = cm
+        magic_mock = MagicMock()
```
Comment (jdkandersson) [inline tests/unit/test_charm.py:55]: Should this describe what is being mocked rather than just its type?

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -68,7 +69,7 @@ def test_charm_reaches_active_status(urlopen_mock):
         )
         harness.begin()
         harness.charm.on.config_changed.emit()
-        harness.model.unit.status == ops.ActiveStatus()
+        assert harness.model.unit.status == ops.ActiveStatus()
```
Comment (jdkandersson) [inline tests/unit/test_charm.py:72]: should we also do this check by name?

File: tests/unit/test_charm.py [hunk 4]
```diff
@@ -80,11 +81,11 @@ def test_relation_joined_when_leader(urlopen_mock, metadata_file):
     assert: the relation get populated with the SAML data.
     """
     with open(f"tests/unit/files/{metadata_file}", "rb") as metadata:
-        cm = MagicMock()
-        cm.getcode.return_value = 200
-        cm.read.return_value = metadata.read()
-        cm.__enter__.return_value = cm
-        urlopen_mock.return_value = cm
+        magic_mock = MagicMock()
```
Comment (jdkandersson) [inline tests/unit/test_charm.py:84]: Similar to above re variable naming

File: tests/unit/test_charm.py [hunk 5]
```diff
@@ -173,7 +174,7 @@ def test_relation_joined_when_not_leader(urlopen_mock):
         )
         harness.begin()
         harness.charm.on.config_changed.emit()
-        harness.model.unit.status == ops.ActiveStatus()
+        assert harness.model.unit.status == ops.ActiveStatus()
```
Comment (jdkandersson) [inline tests/unit/test_charm.py:177]: similar to above

File: tox.ini [hunk 6]
```diff
@@ -43,6 +43,7 @@ deps =
     flake8-docstrings-complete>=1.2.0
     flake8-test-docs>=1.0
     isort
+    mock
```
Comment (jdkandersson) [inline tox.ini:46]: What is the reason for needing this please?

File: tests/unit/test_charm.py (hunk 3 above)
Comment (arturo-seijas) [reply tests/unit/test_charm.py:72]: Not really, this one doesn't contain a status message

File: tests/unit/test_charm.py (hunk 5 above)
Comment (arturo-seijas) [reply tests/unit/test_charm.py:177]: Same here

File: tox.ini (hunk 6 above)
Comment (arturo-seijas) [reply tox.ini:46]: pylist complains about not being able to import this package when linting

File: tests/unit/test_charm.py (hunk 1 above)
Comment (arturo-seijas) [reply tests/unit/test_charm.py:44]: The name is available

File: tests/unit/test_charm.py (hunk 2 above)
Comment (arturo-seijas) [reply tests/unit/test_charm.py]: Indeed

File: tests/unit/test_charm.py [hunk 7]
```diff
@@ -98,7 +99,7 @@ def test_relation_joined_when_leader(urlopen_mock, metadata_file):
...
```
Comment (yanksyoon) [inline tests/unit/test_charm.py:102]: Would it be better if we asserted the `status.name` for consistency with tests above?

File: tests/unit/test_saml.py [hunk 8]
```diff
@@ -2,15 +2,32 @@
 # See LICENSE file for licensing details.
 
 """SAML Integrator unit tests."""
+# pylint: disable=pointless-statement
```
Comment (yanksyoon) [inline tests/unit/test_saml.py:5]: May I ask what this is used for?

File: tests/unit/test_saml.py [hunk 9]
```diff
@@ -42,22 +51,17 @@ def test_saml_with_invalid_metadata(urlopen_mock):
 
 
 @patch.object(urllib.request, "urlopen", side_effect=urllib.error.URLError("Error"))
-def test_saml_with_invalid_url(urlopen_mock):
+def test_saml_with_invalid_url(_):
```
Comment (yanksyoon) [inline tests/unit/test_saml.py:54]: May I ask if the `_` is required as a placeholder for the object patch?

**Review by yanksyoon — APPROVED** [review]
LGTM, minor questions! Thank you!

File: tests/unit/test_saml.py (hunk 8 above)
Comment (arturo-seijas) [reply tests/unit/test_saml.py:5]: For these pieces of code
```
with pytest.raises(CharmConfigInvalidError):
        saml_integrator.certificates
```

File: tests/unit/test_saml.py (hunk 9 above)
Comment (arturo-seijas) [reply tests/unit/test_saml.py:54]: Yes, it is required

File: tests/unit/test_charm.py (hunk 7 above)
Comment (arturo-seijas) [reply tests/unit/test_charm.py:102]: Not necessarily. The only reason we are checking only the name in the assertion above is that the status contains a message

**Review by gtrkiller — APPROVED** [review]
LGTM

File: tox.ini (hunk 6 above)
Comment (jdkandersson) [reply tox.ini:46]: Isn't mock in the inbuilt unittest library? This sounds like something we should investigate - mock shouldn't need to be pip installed

File: tests/unit/test_charm.py (hunk 1 above)
Comment (jdkandersson) [reply tests/unit/test_charm.py:44]: The established pattern in most of our tests is to compare it to `ops.BlockedStatus.name`

**Comment by jdkandersson** [comment]
There was a few comments that should not have been resolved on here yet - I have marked them as unresolved and it would be good to resolve them in a follow up PR please

File: tox.ini (hunk 6 above)
Comment (arturo-seijas) [reply tox.ini:46]: It is, but we are not using unittest here

File: tests/unit/test_charm.py (hunk 1 above)
Comment (arturo-seijas) [reply tests/unit/test_charm.py:44]: Noted

File: tox.ini (hunk 6 above)
Comment (jdkandersson) [reply tox.ini:46]: The `mock` library is a backport to previous version of python of the `unittest.mock` library. Unless you need the backport from a later version of python, importing `unittest.mock` is equivalent to installing `mock` regardless of whether pytest is being used for testing. you can use `unittest.mock` in pytest. Maybe I'm misunderstanding something?

Comment (arturo-seijas) [reply tox.ini:46]: Yes, I can. I think it was me who misunderstood

**Pull request merged**